sql/parser/eval.go

```diff
@@ -581,6 +699,23 @@ func init() {
 	cmpOps[cmpArgs{In, tupleType, tupleType}] = evalTupleIN
 }
 
+// NormalizeAndTypeCheckExpr is a combination of NormalizeExpr and
```

Comment:
Slightly strange that this function is in `eval.go` instead of either `normalize.go` or `type_check.go`.